vsintegration/src/FSharp.Editor/HelpContextService.fs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.VisualStudio.FSharp.Editor
+
+open System
+open System.Threading.Tasks
+open System.Collections.Generic
+open System.Composition
+open Microsoft.CodeAnalysis.Editor
+open Microsoft.CodeAnalysis.Text
+open Microsoft.CodeAnalysis.Classification
+open Microsoft.VisualStudio.FSharp.LanguageService
+open Microsoft.FSharp.Compiler.SourceCodeServices
+open Microsoft.VisualStudio.LanguageServices.Implementation.F1Help
+open Microsoft.CodeAnalysis.Host.Mef
+open Microsoft.FSharp.Compiler.Range
+
+[<Shared>]
+[<ExportLanguageService(typeof<IHelpContextService>, FSharpCommonConstants.FSharpLanguageName)>]
+type internal FSharpHelpContextService 
+    [<ImportingConstructor>]
+    (
+        checkerProvider: FSharpCheckerProvider,
+        projectInfoManager: ProjectInfoManager
+    ) =
+
+    static member GetHelpTerm(checker: FSharpChecker, sourceText : SourceText, fileName, options, span: TextSpan, tokens: List<ClassifiedSpan>, textVersion) = async {
+        let! _parse,check = checker.ParseAndCheckFileInProject(fileName, textVersion, sourceText.ToString(), options)
+        match check with
+        | FSharpCheckFileAnswer.Aborted -> return None
+        | FSharpCheckFileAnswer.Succeeded(check) ->
+            let textLines = sourceText.Lines
+            let lineInfo = textLines.GetLineFromPosition(span.Start)
+            let line = lineInfo.LineNumber
+            let lineText = lineInfo.ToString()
+
+            let caretColumn = textLines.GetLinePosition(span.Start).Character
+
+            let shouldTryToFindSurroundingIdent (token : ClassifiedSpan) =
+                let span = token.TextSpan
+                let content = sourceText.ToString().Substring(span.Start, span.End - span.Start)
+                match token.ClassificationType with
+                | ClassificationTypeNames.Text
+                | ClassificationTypeNames.WhiteSpace -> true
+                | ClassificationTypeNames.Operator when
+                       content = "." -> true
+                | _ -> false
+          
+            let keyword =
```

Comment:
I missed that one, thanks. It's not quite applicable here because it doesn't give back the categories that are needed here. But the code helped me do a small cleanup.